app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,54 @@
+class PicturesController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def create
+    @picture = current_user.pictures.new(picture_params)
+    if @picture.save
+      redirect_to pictures_path
+    else
+      redirect_to new_picture_path
+      flash[:notice] = 'Error: You must upload a picture'
+    end
+  end
+
+  def new
+    @picture = Picture.new
+  end
+
+  def edit
+    @picture = Picture.find(params[:id])
+    unless current_user.pictures.include? @picture
```

Comment:
I'm not expert on these things, but it might be more performant to do:
`@picture.user == current_user` 

The way it is right now is loading all of the users pictures from the DB.